dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -109,6 +114,35 @@ def remove_mapping
     redirect_to @regional_partner
   end
 
+  # POST /regional_partners/:id/replace_mappings
+  # Replace mappings for partner with given id with contents
+  # of CSV file passed in regions parameter
+  # CSV file must have header 'Region' and all regions must be
+  # valid (zip code or state), or no replacement will be done
+  def replace_mappings
+    regions_file = params[:regions]
+    csv = validate_mappings_csv(regions_file)
+    unless csv
+      redirect_to @regional_partner
+      return
+    end
+
+    mappings, errors = parse_mappings_from_csv(csv)
+
+    if !errors.empty?
```

Comment:
Suggestion: Ruby style seems to prefer using `unless errors.empty?` in this case to remove the negation operation. IMO, using `if errors.empty?` (and switching the positive case up) is better because not only it removes negation but also it is easier to read (slight less mental work).